Ok, I've added the missing dependency.

However, we are not quite done yet: :)

```
adora-belle> configure (lib + exe)
Configuring adora-belle-0.0.0...
adora-belle> build (lib + exe)
Preprocessing library for adora-belle-0.0.0..
Building library for adora-belle-0.0.0..
[1 of 2] Compiling App

/home/noctux/projects/adorabelle/src/App.hs:221:40: error:
    Ambiguous occurrence ‘map’
    It could refer to
       either ‘Prelude.map’,
              imported from ‘Prelude’ at src/App.hs:8:8-10
              (and originally defined in ‘GHC.Base’)
           or ‘HM.map’,
              imported from ‘Data.HashMap.Strict’ at src/App.hs:24:1-42
              (and originally defined in ‘unordered-containers-0.2.10.0:Data.HashMap.Strict.Base’)
    |
221 |         return $ ((mapLeft (unlines . (map messageString) . errorMessages) parsed >>=
    |                                        ^^^

/home/noctux/projects/adorabelle/src/App.hs:306:32: error:
    Ambiguous occurrence ‘filter’
    It could refer to
       either ‘Prelude.filter’,
              imported from ‘Prelude’ at src/App.hs:8:8-10
              (and originally defined in ‘GHC.List’)
           or ‘HM.filter’,
              imported from ‘Data.HashMap.Strict’ at src/App.hs:24:1-42
              (and originally defined in ‘unordered-containers-0.2.10.0:Data.HashMap.Base’)
    |
306 | expireLog maxbacklog now log = filter (\LogItem{timeStamp=rtime} -> (diffUTCTime now rtime) < maxdiff) log
    |                                ^^^^^^

/home/noctux/projects/adorabelle/src/App.hs:312:74: error:
    Not in scope: ‘HT.elems’
    Perhaps you meant ‘HM.elems’ (imported from Data.HashMap.Strict)
    No module named ‘HT’ is imported.
    |
312 |   any usermatches $ Prelude.map (\(HelpRequest _ ident _ _) -> ident) $ (HT.elems a) ++ p
    |                                                                          ^^^^^^^^

/home/noctux/projects/adorabelle/src/App.hs:353:25: error:
    Ambiguous occurrence ‘map’
    It could refer to
       either ‘Prelude.map’,
              imported from ‘Prelude’ at src/App.hs:8:8-10
              (and originally defined in ‘GHC.Base’)
           or ‘HM.map’,
              imported from ‘Data.HashMap.Strict’ at src/App.hs:24:1-42
              (and originally defined in ‘unordered-containers-0.2.10.0:Data.HashMap.Strict.Base’)
    |
353 |     let removedactive = map snd $ findActive a
    |                         ^^^

/home/noctux/projects/adorabelle/src/App.hs:354:27: error:
    Ambiguous occurrence ‘foldr’
    It could refer to
       either ‘Prelude.foldr’,
              imported from ‘Prelude’ at src/App.hs:8:8-10
              (and originally defined in ‘Data.Foldable’)
           or ‘HM.foldr’,
              imported from ‘Data.HashMap.Strict’ at src/App.hs:24:1-42
              (and originally defined in ‘unordered-containers-0.2.10.0:Data.HashMap.Base’)
    |
354 |         remainingactive = foldr (\k m -> HT.delete k m) a $ map fst $ findActive a
    |                           ^^^^^

/home/noctux/projects/adorabelle/src/App.hs:354:42: error:
    Not in scope: ‘HT.delete’
    Perhaps you meant ‘HM.delete’ (imported from Data.HashMap.Strict)
    No module named ‘HT’ is imported.
    |
354 |         remainingactive = foldr (\k m -> HT.delete k m) a $ map fst $ findActive a
    |                                          ^^^^^^^^^

/home/noctux/projects/adorabelle/src/App.hs:354:61: error:
    Ambiguous occurrence ‘map’
    It could refer to
       either ‘Prelude.map’,
              imported from ‘Prelude’ at src/App.hs:8:8-10
              (and originally defined in ‘GHC.Base’)
           or ‘HM.map’,
              imported from ‘Data.HashMap.Strict’ at src/App.hs:24:1-42
              (and originally defined in ‘unordered-containers-0.2.10.0:Data.HashMap.Strict.Base’)
    |
354 |         remainingactive = foldr (\k m -> HT.delete k m) a $ map fst $ findActive a
    |                                                             ^^^

/home/noctux/projects/adorabelle/src/App.hs:374:27: error:
    Not in scope: ‘HT.filter’
    Perhaps you meant one of these:
      ‘HM.filter’ (imported from Data.HashMap.Strict),
      ‘BL.filter’ (imported from Data.ByteString.Lazy)
    No module named ‘HT’ is imported.
    |
374 |     findActive = toList $ HT.filter matchesid
    |                           ^^^^^^^^^

/home/noctux/projects/adorabelle/src/App.hs:381:16: error:
    Not in scope: ‘HT.map’
    Perhaps you meant one of these:
      ‘HM.map’ (imported from Data.HashMap.Strict),
      ‘BL.map’ (imported from Data.ByteString.Lazy)
    No module named ‘HT’ is imported.
    |
381 |   let active = HT.map blind a
    |                ^^^^^^

/home/noctux/projects/adorabelle/src/App.hs:407:22: error:
    Ambiguous occurrence ‘map’
    It could refer to
       either ‘Prelude.map’,
              imported from ‘Prelude’ at src/App.hs:8:8-10
              (and originally defined in ‘GHC.Base’)
           or ‘HM.map’,
              imported from ‘Data.HashMap.Strict’ at src/App.hs:24:1-42
              (and originally defined in ‘unordered-containers-0.2.10.0:Data.HashMap.Strict.Base’)
    |
407 |             keys   = map fst active
    |                      ^^^

/home/noctux/projects/adorabelle/src/App.hs:408:47: error:
    Ambiguous occurrence ‘map’
    It could refer to
       either ‘Prelude.map’,
              imported from ‘Prelude’ at src/App.hs:8:8-10
              (and originally defined in ‘GHC.Base’)
           or ‘HM.map’,
              imported from ‘Data.HashMap.Strict’ at src/App.hs:24:1-42
              (and originally defined in ‘unordered-containers-0.2.10.0:Data.HashMap.Strict.Base’)
    |
408 |             logitem =  logtemplate{requests = map snd $ active}
    |                                               ^^^

/home/noctux/projects/adorabelle/src/App.hs:410:20: error:
    Ambiguous occurrence ‘map’
    It could refer to
       either ‘Prelude.map’,
              imported from ‘Prelude’ at src/App.hs:8:8-10
              (and originally defined in ‘GHC.Base’)
           or ‘HM.map’,
              imported from ‘Data.HashMap.Strict’ at src/App.hs:24:1-42
              (and originally defined in ‘unordered-containers-0.2.10.0:Data.HashMap.Strict.Base’)
    |
410 |           (Right $ map snd $ active, s{ activeRequests=HT.filterWithKey (\k v -> k `elem` keys) a, actionLog=expire (logitem:log)})
    |                    ^^^

/home/noctux/projects/adorabelle/src/App.hs:410:56: error:
    Not in scope: ‘HT.filterWithKey’
    Perhaps you meant one of these:
      ‘HM.filterWithKey’ (imported from Data.HashMap.Strict),
      ‘Map.filterWithKey’ (imported from Data.Map),
      ‘HM.foldrWithKey’ (imported from Data.HashMap.Strict)
    No module named ‘HT’ is imported.
    |
410 |           (Right $ map snd $ active, s{ activeRequests=HT.filterWithKey (\k v -> k `elem` keys) a, actionLog=expire (logitem:log)})
    |                                                        ^^^^^^^^^^^^^^^^

/home/noctux/projects/adorabelle/src/App.hs:423:56: error:
    Not in scope: ‘HT.insert’
    Perhaps you meant ‘HM.insert’ (imported from Data.HashMap.Strict)
    No module named ‘HT’ is imported.
    |
423 |                       (Right matched, s{activeRequests=HT.insert (Responder name) x a, pendingRequests = rest, actionLog=expire (logitem:log)})
    |                                                        ^^^^^^^^^

/home/noctux/projects/adorabelle/src/App.hs:433:22: error:
    Ambiguous occurrence ‘map’
    It could refer to
       either ‘Prelude.map’,
              imported from ‘Prelude’ at src/App.hs:8:8-10
              (and originally defined in ‘GHC.Base’)
           or ‘HM.map’,
              imported from ‘Data.HashMap.Strict’ at src/App.hs:24:1-42
              (and originally defined in ‘unordered-containers-0.2.10.0:Data.HashMap.Strict.Base’)
    |
433 |             keys   = map fst active
    |                      ^^^

/home/noctux/projects/adorabelle/src/App.hs:434:22: error:
    Ambiguous occurrence ‘map’
    It could refer to
       either ‘Prelude.map’,
              imported from ‘Prelude’ at src/App.hs:8:8-10
              (and originally defined in ‘GHC.Base’)
           or ‘HM.map’,
              imported from ‘Data.HashMap.Strict’ at src/App.hs:24:1-42
              (and originally defined in ‘unordered-containers-0.2.10.0:Data.HashMap.Strict.Base’)
    |
434 |             reqs   = map snd $ active
    |                      ^^^

/home/noctux/projects/adorabelle/src/App.hs:437:44: error:
    Not in scope: ‘HT.filterWithKey’
    Perhaps you meant one of these:
      ‘HM.filterWithKey’ (imported from Data.HashMap.Strict),
      ‘Map.filterWithKey’ (imported from Data.Map),
      ‘HM.foldrWithKey’ (imported from Data.HashMap.Strict)
    No module named ‘HT’ is imported.
    |
437 |           (Right $ reqs, s{ activeRequests=HT.filterWithKey (\k v -> k `elem` keys) a, pendingRequests=reqs ++ p, actionLog=expire (logitem:log)})
    |                                            ^^^^^^^^^^^^^^^^

/home/noctux/projects/adorabelle/src/App.hs:445:30: error:
    Not in scope: ‘HT.filter’
    Perhaps you meant one of these:
      ‘HM.filter’ (imported from Data.HashMap.Strict),
      ‘BL.filter’ (imported from Data.ByteString.Lazy)
    No module named ‘HT’ is imported.
    |
445 |     findActive    = toList $ HT.filter matchesid
    |                              ^^^^^^^^^


--  While building package adora-belle-0.0.0 using:
      /home/noctux/.local/share/stack/setup-exe-cache/x86_64-linux-tinfo6/Cabal-simple_mPHDZzAJ_3.0.1.0_ghc-8.8.3 --builddir=.stack-work/dist/x86_64-linux-tinfo6/Cabal-3.0.1.0 build lib:adora-belle exe:adora-belle --ghc-options " -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
```

--- a/src/App.hs
+++ b/src/App.hs
@@ -1,5 +1,6 @@
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingVia #-}
 {-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE TypeOperators #-}
 {-# LANGUAGE DuplicateRecordFields #-}
@@ -60,10 +61,10 @@ data HelpRequest = HelpRequest { reqid :: RequestID, userid :: Identification, t
   deriving (Generic, Show, Eq)
 instance ToJSON HelpRequest
 
-data Responder = Responder String
+newtype Responder = Responder String
   deriving (Generic, Show, Eq, Ord)
+  deriving ToJSONKey via String
 instance ToJSON Responder
-instance ToJSONKey Responder
 
 data State = State { lectureName     :: String
                    , timeSlots       :: [TimeRange]

--- a/www/static/adorabelle.js
+++ b/www/static/adorabelle.js
@@ -306,21 +306,16 @@ function formatBacklogItem(item) {
 var seenAdmins = [];
 
 function updateServicedRequests(servicedRequests) {
-	var dict = new Object();
-	servicedRequests.forEach(function (item, index) {
-		dict[item[0]] = item[1];
-	});
-
 	var order;
 	if (admininterface) {
-		Object.keys(dict).forEach(function (elem) {
+		Object.keys(servicedRequests).forEach(function (elem) {
 			if(!seenAdmins.includes(elem)) {
 				seenAdmins.push(elem);
 			}
 		});
 		order = seenAdmins;
 	} else {
-		order = Object.keys(dict).sort();
+		order = Object.keys(servicedRequests).sort();
 	}
 
 	var elem = $('#servicedReqs').clone().empty();
@@ -328,8 +323,8 @@ function updateServicedRequests(servicedRequests) {
 		var lbl = $('<label class="d-flex justify-content-start col-form-label-lg"/>');
 		lbl.text(key);
 		elem.append(lbl);
-		if (key in dict) {
-			elem.append(formatRequest(dict[key], "active"));
+		if (key in servicedRequests) {
+			elem.append(formatRequest(servicedRequests[key], "active"));
 		} else {
 			elem.append(formatRequest(undefined, "dummy"));
 		}

